finetrainers/ltx_video/ltx_video_lora.py

```diff
@@ -174,8 +313,30 @@ def post_latent_preparation(
     width: int,
     patch_size: int = 1,
     patch_size_t: int = 1,
+    image_condition_dropout_p: float = 0.0,
+    image_condition_noise_scale: float = 0.0,
+    image_condition_noise_type: Optional[str] = "gaussian",
+    generator: Optional[torch.Generator] = None,
+    **kwargs,
 ) -> torch.Tensor:
+    assert image_condition_noise_type == "gaussian", "Only Gaussian noise is supported for now for LTX Video"
+
     latents = _normalize_latents(latents, latents_mean, latents_std)
+
+    # We know that the first frame is the image condition frame
+    if image_condition_dropout_p > 0.0:
+        # TODO(aryan): The zero-ing dropout is for testing based on ideas from CogVideoX. Since there is no source to
+        # refer to for LTX Video, this is experimental and should be unsupported if it does not work.
+        if random.random() < image_condition_dropout_p:
+            latents[:, :, 0].zero_()
+        else:
+            # Map from [0, 1] to [0, image_condition_noise_scale]
+            scale_factor = random.random() * image_condition_noise_scale
+            # :/ Because we don't have torch.randn_like
+            latents[:, :, 0] = (
```

Comment:
@yoavhacohen Would love to know your thoughts on this as well. It was an educated guess based on some other training code I've come across for image latent regularisation. Since we're not entirely sure how LTXV was trained, this may probably not be helpful and cause worse results. I'm still experimenting but any details of training for things like this would be super awesome 🤗

Reply:
You should add noise to the conditioning tokens in the same way as the other tokens - just use a different noise level.